plugins/hls-rename-plugin/src/Ide/Plugin/Rename.hs

```diff
@@ -1,188 +1,251 @@
 {-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE FlexibleContexts    #-}
 {-# LANGUAGE GADTs               #-}
 {-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE OverloadedLabels    #-}
+{-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE RankNTypes          #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE TypeApplications    #-}
 
 module Ide.Plugin.Rename (descriptor) where
 
+#if MIN_VERSION_ghc(9,2,1)
+import           GHC.Parser.Annotation                 (AnnContext, AnnList, AnnParen, AnnPragma)
+#endif
+
 import           Control.Monad
-import           Control.Monad.IO.Class               (MonadIO (liftIO))
+import           Control.Monad.IO.Class
 import           Control.Monad.Trans.Class
 import           Control.Monad.Trans.Except
-import           Data.Containers.ListUtils
 import           Data.Generics
-import           Data.List.Extra                      hiding (nubOrd, replace)
-import qualified Data.Map                             as M
+import           Data.HashSet                          (HashSet)
+import qualified Data.HashSet                          as HS
+import           Data.Hashable
+import           Data.List.Extra
+import qualified Data.Map                              as M
 import           Data.Maybe
-import qualified Data.Text                            as T
-import           Development.IDE                      hiding (pluginHandlers)
+import           Data.Mod.Word
+import qualified Data.Text                             as T
 import           Development.IDE.Core.PositionMapping
+import           Development.IDE.Core.RuleTypes
+import           Development.IDE.Core.Service
 import           Development.IDE.Core.Shake
-import           Development.IDE.GHC.Compat
+import           Development.IDE.GHC.Compat.Core
+import           Development.IDE.GHC.Compat.ExactPrint
+import           Development.IDE.GHC.Compat.Parser
+import           Development.IDE.GHC.Compat.Units
+import           Development.IDE.GHC.Error
+import           Development.IDE.GHC.ExactPrint
 import           Development.IDE.Spans.AtPoint
-#if MIN_VERSION_ghc(9,2,1)
-import           GHC.Parser.Annotation                (AnnContext, AnnList,
-                                                       AnnParen, AnnPragma)
-#endif
-#if MIN_VERSION_ghc(9,0,1)
-import           GHC.Types.Name
-#else
-import           Name
-#endif
-import           Development.IDE.GHC.ExactPrint       (GetAnnotatedParsedSource (GetAnnotatedParsedSource))
+import           Development.IDE.Types.Location
 import           HieDb.Query
 import           Ide.Plugin.Config
+import           Ide.Plugin.Properties
 import           Ide.PluginUtils
 import           Ide.Types
-import           Language.Haskell.GHC.ExactPrint
 import           Language.LSP.Server
 import           Language.LSP.Types
 
+instance Hashable Location
+instance Hashable Range
+instance Hashable Position
+instance Hashable UInt
+instance Hashable (Mod a) where hash n = hash (unMod n)
```

Comment:
Where are these orphaned instances used? If not used please delete them, and if used they need to go into one of the Orphan modules in ghcide.

Reply:
They’re used since I changed the list of name references to a HashSet for better lookup, although it didn’t have much of an impact on performance.

[I’ve moved them to Development/IDE/GHC/Orphans.hs](https://github.com/haskell/haskell-language-server/pull/2809/commits/bb91db031e7c1962084d9c69aa3761e2cd118b97)